File: composeApp/src/androidMain/kotlin/com/dosystem/todo/ui/weather/DayForecast.kt
package com.dosystem.todo.ui.weather

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import coil.compose.AsyncImage
import com.dosystem.todo.getUnitAnnotatedString
import com.dosystem.todo.viewmodel.DayForecastUiModel

@Composable
fun DayForecastContent(
    modifier: Modifier = Modifier,
    dayForecasts: List<DayForecastUiModel>
) {
    Column(
        verticalArrangement = Arrangement.spacedBy(12.dp),
        modifier = modifier
            .padding(horizontal = 16.dp)
            .background(Color(0x85858585), RoundedCornerShape(15.dp))
            .padding(20.dp)
    ) {
        for (i in dayForecasts.indices) {
            DayForecastItem(forecast = dayForecasts[i])
        }
    }
}

@Composable
fun DayForecastItem(
    modifier: Modifier = Modifier,
    forecast: DayForecastUiModel
) {
    Row(
        verticalAlignment = Alignment.CenterVertically,
        modifier = modifier.fillMaxWidth()
    ) {
        Text(
            text = forecast.dayOfWeek,
            color = when (forecast.dayOfWeek) {
                "토" -> Color.Blue
                "일" -> Color.Red
                else -> Color.White
            },
            fontWeight = if (forecast.dayOfWeek == "오늘") FontWeight.Bold else FontWeight.Medium
        )
        Spacer(modifier.weight(1f))
        AsyncImage(
            model = forecast.iconUrl,
            contentDescription = null
        )
        Spacer(modifier.width(20.dp))
        Text(
            text = getUnitAnnotatedString("${forecast.maxTemp}  /  ${forecast.minTemp}", unitSize = 12.sp),
            color = Color.White,
            fontWeight = if (forecast.dayOfWeek == "오늘") FontWeight.Bold else FontWeight.Medium
        )
    }
}

File: iosApp/iosApp/Ui/Weather/DayForecast.swift
//
//  DayForecast.swift
//  iosApp
//
//  Created by openobject2 on 9/19/25.
//

import SwiftUI
import Shared

struct DayForecastContent: View {
    let dayForecasts: [DayForecastUiModel]
    
    init(dayForecasts: [DayForecastUiModel]) {
        self.dayForecasts = dayForecasts
    }
    
    var body: some View {
        VStack(spacing: 12) {
            ForEach(dayForecasts, id: \.dayOfWeek) { forecast in
                DayForecastItem(forecast: forecast)
            }
        }
        .padding(20)
        .background(
            RoundedRectangle(cornerRadius: 15).fill(Color(red: 0.52, green: 0.52, blue: 0.52, opacity: 0.52))
        )
        .padding(.horizontal, 16)
        .fixedSize(horizontal: false, vertical: true)
    }
}


struct DayForecastItem: View {
    let forecast: DayForecastUiModel
    private let dayOfWeekColor: Color
    private let dayOfWeekWeight: Font.Weight
    
    init(forecast: DayForecastUiModel) {
        self.forecast = forecast
        
        dayOfWeekColor = switch forecast.dayOfWeek {
        case "일": .red
        case "토": .blue
        default: .white
        }
        dayOfWeekWeight = if forecast.dayOfWeek == "오늘" {
            Font.Weight.bold
        } else {
            Font.Weight.medium
        }
    }
    
    var body: some View {
        HStack {
            Text(forecast.dayOfWeek).foregroundStyle(dayOfWeekColor).fontWeight(dayOfWeekWeight)
            Spacer()
            AsyncImage(url: URL(string: forecast.iconUrl)) { image in
                image.image?.resizable().scaledToFit()
            }
            .frame(width: 50, height: 50)
            .padding(.trailing, 20)
            getUnitAttributedText(prev: "\(forecast.maxTemp)  /  \(forecast.minTemp)", unitSize: 12)
                .foregroundColor(.white)
                .fontWeight(dayOfWeekWeight)
        }
    }
}
